pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -460,8 +623,7 @@ func (nim *nodeInfoManager) installDriverToCSINodeInfo(
 	for _, driverInfoStatus := range nodeInfo.Status.Drivers {
 		if driverInfoStatus.Name == driverName {
 			if driverInfoStatus.Available &&
-				/* TODO(https://github.com/kubernetes/enhancements/issues/625): Add actual migration status */
-				driverInfoStatus.VolumePluginMechanism == csiv1alpha1.VolumePluginMechanismInTree {
+				driverInfoStatus.VolumePluginMechanism == volumePluginMechanism {
 				statusModified = false
 			}
```

Comment:
should there be an `else` block here that does this?
```
driverInfoStatus.Available = true
driverInfoStatus.VolumePluginMechanism = volumePluginMechanism
newDriverStatuses = append(newDriverStatuses, driverInfoStatus)
statusModified = true
```

that would preserve the status order, which is nice

Reply:
same question for spec

Reply:
ah I see what you're saying. I can clean this up a bit

Reply:
I changed it but ended up reverting. The amount of flags made it a little confusing. There was "statusModified" and "specModified" as well as "statusExists" and "specExists" since we need to cover the case where the driver is not currently represented in CSINodeInfo at all.

Reply:
fair enough